src/main/java/appeng/util/item/FluidList.java

```diff
@@ -0,0 +1,219 @@
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.add( option );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public IAEFluidStack findPrecise( IAEFluidStack i )
+	{
+		if( this.isStackNull( i ) )
+		{
+			return null;
+		}
+
+		final IAEFluidStack is = this.getFluidRecord( i );
+
+		return is;
+	}
+
+	@Override
+	public Collection<IAEFluidStack> findFuzzy( IAEFluidStack filter, FuzzyMode fuzzy )
+	{
+		if( this.isStackNull( filter ) )
+		{
+			return Collections.emptyList();
+		}
+
+		final List<IAEFluidStack> result = Lists.newArrayList();
+
+		if( filter.equals( this ) )
+		{
+			result.add( filter );
+		}
+
+		return result;
+
+	}
+
+	@Override
+	public boolean isEmpty()
+	{
+		return !this.iterator().hasNext();
+	}
+
+	@Override
+	public void addStorage( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.incStackSize( option.getStackSize() );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		this.putFluidRecord( opt );
+	}
+
+	/*
+	 * public synchronized void clean() { Iterator<StackType> i = iterator(); while (i.hasNext()) { StackType AEI =
+	 * i.next(); if ( !AEI.isMeaningful() ) i.remove(); } }
+	 */
+
+	@Override
+	public void addCrafting( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.setCraftable( true );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		opt.setStackSize( 0 );
+		opt.setCraftable( true );
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public void addRequestable( IAEFluidStack option )
+	{
+		if( this.isStackNull( option ) )
+		{
+			return;
+		}
+
+		final IAEFluidStack st = this.getFluidRecord( option );
+
+		if( st != null )
+		{
+			st.setCountRequestable( st.getCountRequestable() + option.getCountRequestable() );
+			return;
+		}
+
+		final IAEFluidStack opt = option.copy();
+		opt.setStackSize( 0 );
+		opt.setCraftable( false );
+		opt.setCountRequestable( opt.getCountRequestable() );
+
+		this.putFluidRecord( opt );
+	}
+
+	@Override
+	public IAEFluidStack getFirstItem()
+	{
+		for( IAEFluidStack stackType : this )
+		{
+			return stackType;
+		}
+		return null;
+	}
+
+	@Override
+	public int size()
+	{
+		return this.records.values().size();
+	}
+
+	@Override
+	public Iterator<IAEFluidStack> iterator()
+	{
+		return new MeaningfulFluidIterator<IAEFluidStack>( this.records.values().iterator() );
+	}
+
+	@Override
+	public void resetStatus()
+	{
+		for( IAEFluidStack i : this )
+		{
+			i.reset();
+		}
+	}
+
+	private IAEFluidStack getFluidRecord( IAEFluidStack fluid )
+	{
+		final IAEFluidStack fluidrecord = this.records.get( fluid );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "fluidrecord". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)